numpy/core/arrayprint.py

```diff
@@ -216,18 +213,20 @@ def get_printoptions():
     set_printoptions, set_string_function
 
     """
-    d = dict(precision=_float_output_precision,
-             threshold=_summaryThreshold,
-             edgeitems=_summaryEdgeItems,
-             linewidth=_line_width,
-             suppress=_float_output_suppress_small,
-             nanstr=_nan_str,
-             infstr=_inf_str,
-             formatter=_formatter)
-    return d
+    return dict(precision=_options['_float_output_precision'],
```

Comment:
Maybe just return a deep copy of the `_options` dictionary?

Reply:
since they are strings, simple copy works too.

Reply:
Will do. thanks a lot.

Reply:
The deepcopy avoids having the options change while they are being used. A simple copy of the dictionary can have its contents changed by other functions.